android/app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java

```diff
@@ -199,13 +205,16 @@ private void updateSession() {
         text_subtitle.setText(session.getSubtitle());
         text_track.setText(trackName);
 
-        String date = ISO8601Date.getTimeZoneDateString(
-                ISO8601Date.getDateObject(session.getStartTime())).split(",")[0] + ","
-                + ISO8601Date.getTimeZoneDateString(ISO8601Date.getDateObject(session.getStartTime())).split(",")[1];
-        String startTime = ISO8601Date.getTimeZoneDateString(ISO8601Date.getDateObject(session.getStartTime())).split(",")[2] + ","
-                + ISO8601Date.getTimeZoneDateString(ISO8601Date.getDateObject(session.getStartTime())).split(",")[3];
-        String endTime = ISO8601Date.getTimeZoneDateString(ISO8601Date.getDateObject(session.getEndTime())).split(",")[2] + ","
-                + ISO8601Date.getTimeZoneDateString(ISO8601Date.getDateObject(session.getEndTime())).split(",")[3];
+        simpleDateFormat.setTimeZone(eventTimeZone);
+        String date = null;
+        String startTime = null;
+        String endTime = null;
+
+        if(session.getStartTime()!=null) {
+            date = ISO8601Date.getTimeZoneDateStringFromString(session.getStartTime() + "Z", simpleDateFormat, eventTimeZone).split(",")[0] + "," + ISO8601Date.getTimeZoneDateStringFromString(session.getStartTime() + "Z", simpleDateFormat, eventTimeZone).split(",")[1];
```

Comment:
This can be simplified. The issue I created was to handle these complex formatting in a Util class, so the views just have to call a single method to get a formatted date